js/models/profile/FixedFee.js

```diff
@@ -0,0 +1,38 @@
+import BaseModel from '../BaseModel';
+import app from '../../app';
+import { getListofCurrencies } from '../../data/currencies';
+import is from 'is_js';
+
+export default class extends BaseModel {
+  defaults() {
+    return {
+      currencyCode: 'BTC',
+      amount: 0,
+    };
+  }
+
+  validate(attrs) {
+    const errObj = {};
+    const addError = (fieldName, error) => {
+      errObj[fieldName] = errObj[fieldName] || [];
+      errObj[fieldName].push(error);
+    };
+
+    if (is.not.existy(attrs.currencyCode) || typeof attrs.currencyCode !== 'string') {
+      addError('feeType', app.polyglot.t('settings.moderationTab.moderatorModelErrors.noCurrency'));
+    }
+
+    // make sure currency is a valid value
+    if (getListofCurrencies().indexOf(attrs.currencyCode) === -1) {
```

Comment:
Why can't we use getCurrencyByCode() here?

If the only reason that we created getListofCurrencies() was to satisfy a test, I don't that's a good reason to bring it in, as there's cleaner ways to satisfy the test. I'll write more about that in a bit.